pkg/service/service.go

```diff
@@ -1367,39 +1367,57 @@ func (s *Service) restoreBackendsLocked(svcBackendsById map[lb.BackendID]struct{
 		return fmt.Errorf("Unable to dump backend maps: %s", err)
 	}
 
+	svcBackendsCount := len(svcBackendsById)
 	for _, b := range backends {
 		log.WithFields(logrus.Fields{
 			logfields.BackendID:        b.ID,
 			logfields.L3n4Addr:         b.L3n4Addr.String(),
 			logfields.BackendState:     b.State,
 			logfields.BackendPreferred: b.Preferred,
 		}).Debug("Restoring backend")
-		if _, ok := svcBackendsById[b.ID]; !ok && s.backendRefCount[b.L3n4Addr.Hash()] != 0 {
-			// If a backend by id isn't referenced by any of the service entries,
-			// it's likely to be a duplicate backend. This can happen when agent
-			// leaked backend entries in the backends map prior to restart, and created
-			// duplicate with different IDs but same L3n4Addr (hash).
+		if _, ok := svcBackendsById[b.ID]; !ok && (svcBackendsCount != 0) {
```

Comment:
I'm struggling to interpret the `svcBackendsCount != 0` check. Perhaps an answer to my question above might help. To me it seems that the check will be always true, as there will be at least one service in any K8s cluster (kube-apiserver svc).

Reply:
I've added `(svcBackendsCount != 0)` as a conservative check. In case of unusual cases, if there was a problem reading entries from the services map, we don't want to wipe out the backends map. As this could result in connectivity disruption for existing connections.
Anyway if we don't anticipate extreme cases, it might be okay to drop the check.

Reply:
:+1: It's fine to keep the check, just then let's add a comment saying something similar along these lines:
> In case of unusual cases, if there was a problem reading entries from the services map, we don't want to wipe out the backends map.

Reply:
Done